pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/AggregationGroupByResult.java

```diff
@@ -51,6 +51,15 @@ public Iterator<GroupKeyGenerator.GroupKey> getGroupKeyIterator() {
     return _groupKeyGenerator.getGroupKeys();
   }
 
+  /**
+   * Clear and trim DictionaryBasedGroupKeyGenerator after use
+   */
+  public void clearAndTrimGroupKeyGenerator() {
```

Comment:
We don't need to make it so specific. We can make this class extend `AutoCloseable` and model this as cleanup for `close()`

Reply:
It can implement `AutoClosable` and we can decide either to use `try-with-resource` or manually `close()` based on the caller's context